src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1305,6 +1308,28 @@ private static void RemoveLessDerivedMembers<TMember>(ArrayBuilder<MemberResolut
             }
         }
 
+        private void RemoveLessSpecificExtensionMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMember>> results, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+            where TMember : Symbol
+        {
+            for (int i = 0; i < results.Count; ++i)
+            {
+                var result = results[i];
+
+                if (result.Result.IsValid || result.HasUseSiteDiagnosticToReport)
+                {
+                    if (result.Member.ContainingType.ExtendedTypeNoUseSiteDiagnostics is not { } extendedType)
+                    {
+                        continue;
+                    }
+
+                    if (IsLessSpecificThanAnyExtensionMember(index: i, extendedType, results, ref useSiteInfo))
+                    {
+                        results[i] = result.WithResult(MemberAnalysisResult.LessDerived());
```

Comment:
Should this be a new `MemberAnalysisResult` rather than `LessDerived`?

Reply:
I could not find a benefit to having a separate `MemberResolutionKind`. We're trying to emulate hiding as-if the extension members belonged to the type hierarchy, so I feel okay keeping a single concept. 
I'll add a comment in the enum to point out the overloaded meaning.